templates/base/client/.rtag/src/main/java/rtag/RtagClient.java.hbs

```diff
@@ -0,0 +1,163 @@
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+
+import org.java_websocket.client.WebSocketClient;
+import org.java_websocket.handshake.ServerHandshake;
+import org.msgpack.jackson.dataformat.MessagePackFactory;
+
+import lombok.Setter;
+
+{{#each methods}}
+import rtag.Types.{{makeRequestName @key}};
+{{/each}}
+import rtag.Types.Result;
+import rtag.Types.ResultCallback;
+import rtag.Types.StateChangeRunnable;
+import rtag.Types.Message;
+import rtag.Types.ResponseMessage;
+import rtag.Types.StateMessage;
+import rtag.Types.LoginResponse;
+import rtag.Types.CreateStateResponse;
+import rtag.Types.UserData;
+
+public class RtagClient {
+    private Map<String, ResultCallback> callbacks = new HashMap<String, ResultCallback>();
+    private static ExecutorService es = Executors.newSingleThreadExecutor();
+    private static OkHttpClient client = new OkHttpClient();
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    private static ObjectMapper jsonMapper = new ObjectMapper();
+    private static ObjectMapper msgPackMapper = new ObjectMapper(new MessagePackFactory());
+
+    @Setter
+    private WebSocketClient socket;
+
+    private RtagClient() {
+    }
+
+    public static UserData getUserFromToken(String token) throws IOException {
+        JWT.decode(token);
+        byte[] data = Base64.getDecoder().decode(JWT.decode(token).getPayload());
+        return UserData.builder().value(jsonMapper.readValue(data, UserData.Base.class)).build();
+    }
+
+    {{#each auth}}
+    {{#if (eq @key "anonymous")}}
+    public static String loginAnonymous(String origin) throws IOException {
+        Request request = new Request.Builder().url(String.format("%s/login/anonymous", origin))
+                .post(RequestBody.create(JSON, "{}")).build();
+        String response = client.newCall(request).execute().body().string();
+        return jsonMapper.readValue(response, LoginResponse.class).getToken();
+    }
+
+    {{else if (eq @key "google")}}
+    public static String loginGoogle(String origin) throws IOException {
+        Request request = new Request.Builder().url(String.format("%s/login/google", origin))
+                .post(RequestBody.create(JSON, "{}")).build();
+        String response = client.newCall(request).execute().body().string();
+        return jsonMapper.readValue(response, LoginResponse.class).getToken();
+    }
+
+    {{/if}}
+    {{/each}}
+
+    public static String createState(String origin, String token, ICreateGameRequest request) throws IOException {
+        String response = client.newCall(new Request.Builder().url(String.format("%s/new", origin))
+                .post(RequestBody.create(JSON, "{}")).header("Authorization", "Bearer " + token).build()).execute()
+                .body().string();
+        return jsonMapper.readValue(response, CreateStateResponse.class).getStateId();
```

Comment:
do you think you can make a static helper method for the http calls to clean them up a bit?